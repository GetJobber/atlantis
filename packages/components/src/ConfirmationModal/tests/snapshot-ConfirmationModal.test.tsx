/**
 * Once this PR lands we should be able to un-split these different test files.
 * https://github.com/GetJobber/atlantis/pull/171/files
 */

import React, { ReactPortal } from "react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { ConfirmationModal } from "..";

beforeEach(() => {
  ReactDOM.createPortal = jest.fn(element => {
    return element as ReactPortal;
  });
});

afterEach(() => {
  (ReactDOM.createPortal as jest.Mock).mockClear();
});

it("renders a simple ConfirmationModal", () => {
  const tree = renderer
    .create(
      <ConfirmationModal
        title="Should we?"
        message="Do somethingâ€¦"
        open={true}
        confirmLabel="We Shall"
        onConfirm={() => jest.fn()}
        onCancel={() => jest.fn()}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
