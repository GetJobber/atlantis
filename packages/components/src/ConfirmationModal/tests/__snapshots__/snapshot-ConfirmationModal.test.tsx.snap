// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a simple ConfirmationModal 1`] = `
<div
  className="container"
  tabIndex={0}
>
  <div
    className="overlay"
    style={
      Object {
        "opacity": 0,
      }
    }
  />
  <div
    className="modal small"
    style={
      Object {
        "opacity": 0,
        "transform": "scale(0.9) translateZ(0)",
      }
    }
  >
    <div
      className="header"
      data-testid="modal-header"
    >
      <h3
        className="base extraBold large uppercase"
      >
        Should we?
      </h3>
    </div>
    <div
      className="padded base"
    >
      <div
        className="padded base"
      >
        <p
          className="base regular base text"
        >
          Do somethingâ€¦
        </p>
      </div>
    </div>
    <div
      className="actionBar"
    >
      <div
        className="rightAction"
      >
        <button
          className="button base work primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="base extraBold small uppercase"
          >
            We Shall
          </span>
        </button>
        <button
          className="button base cancel secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="base extraBold small uppercase"
          >
            Cancel
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
